hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -3858,26 +3864,30 @@ void reportProcedureDone(ReportProcedureDoneRequest request) throws IOException
   }
 
   private RegionServerStatusService.BlockingInterface ensureRssStub() {
-    return ensureRssStub(false, Integer.MAX_VALUE);
+    return ensureRssStub(0);
   }
 
-  private RegionServerStatusService.BlockingInterface ensureRssStub(
-      boolean checkClusterConn, int maxRetries) {
+  private RegionServerStatusService.BlockingInterface ensureRssStub(int maxRetries) {
     RegionServerStatusService.BlockingInterface rss = null;
-    while (maxRetries-- >= 0 && (!checkClusterConn
-      || (checkClusterConn && clusterConnection != null && !clusterConnection.isClosed()))) {
-      rss = rssStub;
-      if (rss != null) {
-        break;
+    while (rss == null && maxRetries-- >= 0
+        && (clusterConnection != null && !clusterConnection.isClosed())) {
+      synchronized (rssStubLock) {
+        if (resetRssStub) {
```

Comment:
I do not fully understand what's the usage of this `resetRssStub` flag... We do not test it in the condition of the while loop?